File: src/test/cpp/uml/operationTest.cpp
#include "gtest/gtest.h"
#include "uml/operation.h"
#include "uml/opaqueBehavior.h"

using namespace UML;

class OperationTest : public ::testing::Test {};

TEST_F(OperationTest, AddParameterTest) {
    Operation o;
    Parameter p;
    ASSERT_NO_THROW(o.addParameter(p));
    ASSERT_TRUE(o.parameters.front()->uuid == p.uuid);
}

TEST_F(OperationTest, RemoveParameterTest) {
    Operation o;
    Parameter p;
    Parameter p2;
    o.addParameter(p);
    o.addParameter(p2);
    ASSERT_TRUE(o.parameters.front()->uuid == p.uuid);
    ASSERT_NO_THROW(o.removeParameter(p));
    ASSERT_TRUE(o.parameters.front()->uuid == p2.uuid);
}

TEST_F(OperationTest, AddMethodTest) {
    Operation o;
    OpaqueBehavior b;
    ASSERT_NO_THROW(o.addMethod(b));
    ASSERT_TRUE(o.methods.front()->uuid == b.uuid);
}

TEST_F(OperationTest, RemoveMethodTest) {
    Operation o;
    OpaqueBehavior b;
    OpaqueBehavior b2;
    o.addMethod(b);
    o.addMethod(b2);
    ASSERT_TRUE(o.methods.front()->uuid == b.uuid);
    ASSERT_NO_THROW(o.removeMethod(b));
    ASSERT_TRUE(o.methods.front()->uuid == b2.uuid);
}

File: src/test/python/uml/operation_test.py
import unittest
from yuml_python import *

class OperationTest(unittest.TestCase):

    def testGetAndSetType(self):
        t = Classifier()
        o = Operation()
        o.setType(t)
        self.assertEqual(t.getID(), o.getType().getID())

    def testAddAndRemoveParameter(self):
        p = Parameter()
        o = Operation()
        o.addParameter(p)
        self.assertEqual(o.parameters[0].getID(), p.getID())
        o.removeParameter(p)
        self.assertEqual(len(o.parameters) , 0)

if __name__ == '__main__':
    unittest.main()